### ISSUE TITLE
Nesting level restriction

### ISSUE THREAD

**Issue by mmd-osm** [drolbr/Overpass-API#327]
The following query triggers the error message "parse error: Nesting of statements is limited to 1023 levels ", although we only have a single union statement in there:

http://overpass-turbo.eu/s/jpd

This seems to be an issue in this line: https://github.com/drolbr/Overpass-API/blob/b32ba62b126c6cc102ae171ebc7607bb45479554/src/overpass_api/frontend/map_ql_parser.cc#L275

`depth` is incremened in a way that it doesn't represent the nesting level in a sense of one method calling another method. Originally, the purpose of this check was to avoid a too deep method nesting, causing stack overflows.

See https://github.com/mvexel/overpass-api-python-wrapper/issues/66

Also, the XML representation doesn't really have a 'nesting' of that many levels:

```
<osm-script>
  <union into="_">
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>
    <id-query into="_" ref="10" type="way"/>

  </union>
</osm-script>
```

**Comment by drolbr** [comment]
Fixed in 4ad44ad491b12bce24f177d6f2920048aa2ea53c

**Comment by mmd-osm** [comment]
Fix doesn't seem to be available on overpass-api.de, although it is in minor issues branch...
